relay-server/src/metrics_extraction/transactions.rs

```diff
@@ -199,10 +215,50 @@ fn keep_transaction_name(source: &TransactionSource) -> bool {
     }
 }
 
+/// Decide whether we want to keep the transaction name.
+/// High-cardinality sources are excluded to protect our metrics infrastructure.
+/// Note that this will produce a discrepancy between metrics and raw transaction data.
+fn get_transaction_name(
+    event: &Event,
+    accept_transaction_names: &AcceptTransactionNames,
+) -> Option<String> {
+    let original_transaction_name = match event.transaction.value() {
+        Some(name) => name,
+        None => {
+            return None;
+        }
+    };
+
+    let use_original_name = match accept_transaction_names {
+        AcceptTransactionNames::ClientBased => {
+            let sdk_name = event
+                .client_sdk
+                .value()
+                .and_then(|sdk| sdk.name.value())
+                .map(|s| s.as_str())
+                .unwrap_or_default();
+
+            (sdk_name != "sentry.javascript.react" && sdk_name != "sentry.javascript.browser")
+                || is_low_cardinality(event.get_transaction_source())
+        }
+        AcceptTransactionNames::Strict => is_low_cardinality(event.get_transaction_source()),
+    };
+
+    if use_original_name {
+        Some(original_transaction_name.clone())
+    } else {
+        // Pick a sentinel based on the transaction source:
+        match event.get_transaction_source() {
+            TransactionSource::Unknown | TransactionSource::Other(_) => None,
```

Comment:
We could avoid edge-cases and dealing with `null` if we add an explicit placeholder here rather than `null`.

@k-fish do you have preferences for using this in the performance pages?

Reply:
Let me take that back. If we keep null here, this will correspond with `null` in the indexed transaction source, too.